File: barberia/src/Services/UserService.js
import axios from "axios";

const url='http://localhost:8080';

class UserService{

    addUser(user){
        return axios.post(url+'/api/v2/addUser',user) 
    }

     loginUser(data){
        return axios.post(url+'/api/v2/login',data)
     }

     getUserByEmail(email){
        return axios.get(url+'/api/v2/findByEmail/'+email)
     }

     updateUser(d){
        return axios.put(url+'/api/v2/updateUserProfile/{userid}',d)
     }
  }

export default new UserService();

File: Barberia_BackEnd/src/main/java/com/barberia/service/UserService.java
package com.barberia.service;

import java.util.List;

import com.barberia.model.UserRegister;

public interface UserService {
	
	public UserRegister addUser(UserRegister userRegister);
	
	public UserRegister login(UserRegister userRegister);
	
	public UserRegister findUserById(int userid);
	
	public List<UserRegister>getAllUser();
	
	public UserRegister updateUserProfile(UserRegister userRegister);
			
	public UserRegister findByEmail(String email);
}
